deploy/podman/README.md

```diff
@@ -1,24 +1,74 @@
 
-## Requirements
+## General Requirements
 
-Make sure you have [podman](https://podman.io) version 3.3+ installed. If you
+Make sure you have [Podman](https://podman.io) version 3.3+ installed. If you
 must use an older version of podman, reference the [previous documentation and
 procedure](https://github.com/openshift/assisted-service/tree/v2.0.11#deploy-without-a-kubernetes-cluster)
 to avoid a [podman bug](https://github.com/containers/podman/issues/9609).
 
-Grab [pod.yml](./pod.yml) and [configmap.yml](configmap.yml) from this
-directory. No need to clone the whole repo.
+## Requirements for an HTTP based deployment
+
+### Files
+
+No need to clone the whole repo. Use:
+* [pod.yml](./pod.yml)
+* [configmap.yml](configmap.yml)
+
+### Environment Variables
 
 Change `IMAGE_SERVICE_BASE_URL` and `SERVICE_BASE_URL` in
 [configmap.yml](./configmap.yml) to match the hostname or IP address of your
 host. For example if your IP address is 192.168.122.2, then the
 `SERVICE_BASE_URL` would be set to <http://192.168.122.2:8090>. Port 8090 is
 the assisted-service API that agents will connect to.
 
+## Requirements for an HTTPS based deployment
+
+* Assisted UI supports HTTPS starting from UI version [v2.10.0](https://github.com/openshift-assisted/assisted-ui/releases/tag/v2.10.0)
+
+### SSL Certificates
+
+:warning: This example uses self-signed certificates for simplicity, substitute in valid certificates if they are available. Note that self-signed certificates are **not recommended** for use in production.
+
+#### Generate
+
+```shell
+openssl req -x509 -sha256 -nodes -days 365 -newkey rsa:2048 -keyout tls.key -out tls.crt -subj "/CN=localhost" -addext 'subjectAltName = IP:127.0.0.1'
+```
+#### Embed to a ConfigMap
+
+Embed both key and certificate in [configmap_ssl_certs.yml](configmap_ssl_certs.yml)
+Those will get mounted to `assisted-installer-service`, `assisted-installer-image-service` and `assisted-installer-ui` for encryption.
+
+### Files
+
+No need to clone the whole repo. Use:
+* [pod_ssl.yml](./pod_ssl.yml)
+* [configmap_ssl.yml](configmap_ssl.yml)
+* [configmap_ssl_certs.yml](configmap_ssl_certs.yml)
+
+### Environment Variables
+
+Change `IMAGE_SERVICE_BASE_URL` and `SERVICE_BASE_URL` in
+[configmap_ssl.yml](./configmap_ssl.yml) to match the hostname or IP address of your
+host. For example if your IP address is 192.168.122.2, then the
+`SERVICE_BASE_URL` would be set to <https://192.168.122.2:8090>. Port 8090 is
+the assisted-service API that agents will connect to.
+
+Additionally, make sure to change the following:
+```yaml
+ASSISTED_SERVICE_URL: https://127.0.0.1:8090  
+HTTPS_CERT_FILE: "/etc/certs/tls.crt" # Need to match certs-configmap-volume mountPath
+HTTPS_KEY_FILE: "/etc/certs/tls.key"  # Need to match certs-configmap-volume mountPath
+HTTPS_CA_FILE: "/etc/certs/tls.crt"   # Need to match certs-configmap-volume mountPath. Needed if the certificate is not signed by a valid CA.
```

Comment:
By "valid CA", do you mean one that appears in the RHEL trusted cert bundle?

Reply:
It is possible that a user might have their own valid CA, no? 
My intention here is that this is needed for cases where the files with signed by a proper CA and were not self-signed.

If the wording here is not reflecting that, I'll change it. WDYT?

Reply:
I'm looking for a more precise word or phrase than "proper" or "valid". What would make a CA "valid" or "proper" in your scenario?

Reply:
I changed it to what you proposed in our slack discussion.